<a href="https://colab.research.google.com/github/Near-o7/desafio1_dataScience/blob/main/AluraStoreLatam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importación de datos



In [10]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

In [ ]:
pd.set_option('display.float_format', '{:,.2f}'.format)

In [35]:
tienda[["Precio", "Costo de envío"]].agg(["count", "mean", "sum", "min", "max"])

Precio  Costo de envío
count         2,359.00        2,359.00
mean        487,867.91       26,018.61
sum   1,150,880,400.00   61,377,900.00
min           7,600.00            0.00
max       2,977,000.00      160,800.00

In [33]:
tienda2[["Precio", "Costo de envío"]].agg(["count", "mean", "sum", "min", "max"])

Precio  Costo de envío
count         2,359.00        2,359.00
mean        473,227.43       25,216.24
sum   1,116,343,500.00   59,485,100.00
min           7,600.00            0.00
max       2,953,400.00      162,500.00

In [32]:
tienda3[["Precio", "Costo de envío"]].agg(["count", "mean", "sum", "min", "max"])

Precio  Costo de envío
count         2,359.00        2,359.00
mean        465,459.77       24,805.68
sum   1,098,019,600.00   58,516,600.00
min           7,600.00            0.00
max       2,968,200.00      158,400.00

In [31]:
tienda4[["Precio", "Costo de envío"]].agg(["count", "mean", "sum", "min", "max"])

Precio  Costo de envío
count         2,358.00        2,358.00
mean        440,362.89       23,459.46
sum   1,038,375,700.00   55,317,400.00
min           7,600.00            0.00
max       2,902,200.00      154,700.00

En primera instancia podemos observar algunos indicadores:
---
**Ranking por total de ventas**
1.   Tienda1 1,150,880,400
2.   Tienda2 1,116,343,500
3.   Tienda3 1,098,019,600.00
4.   Tienda4 1,038,375,700.00
---
**Ranking por precio promedio de ventas**

1.   Tienda1 487,867
2.   Tienda2 473,227
3.   Tienda3 465,459
4.   Tienda4 440,362




#1. Análisis de facturación



In [42]:
comparacion = pd.DataFrame({
    "Tienda": ["tienda", "tienda2", "tienda3", "tienda4"],
    "Suma Precio": [
        tienda["Precio"].sum(),
        tienda2["Precio"].sum(),
        tienda3["Precio"].sum(),
        tienda4["Precio"].sum()
    ]
})
comparacion = comparacion.sort_values(by="Suma Precio", ascending=False)
comparacion["Suma Precio"] = comparacion["Suma Precio"].apply(lambda x: "{:,.0f}".format(x))
print(comparacion)

    Tienda    Suma Precio
0   tienda  1,150,880,400
1  tienda2  1,116,343,500
2  tienda3  1,098,019,600
3  tienda4  1,038,375,700


Al analizar y comparar las ventas totales utilizando la columna Precio, observamos que la Tienda 1 fue la que obtuvo la mayor facturación, seguida por la Tienda 2, la Tienda 3 y, finalmente, la Tienda 4. Este resultado es consistente con el análisis exploratorio realizado al inicio.

# 2. Ventas por categoría

# 3. Calificación promedio de la tienda


# 4. Productos más y menos vendidos

# 5. Envío promedio por tienda